pkg/hostap/monitor.go

```diff
@@ -240,36 +257,58 @@ func (m *InterfaceMonitor) handleIfEvents() {
 			close(m.ifEventCh)
 			return
 		case linkUpdateEvent := <-m.ifEventCh:
-			newOpState := linkUpdateEvent.Link.Attrs().OperState
-			if newOpState == m.operState {
-				level.Info(m.Logger).Log("interface", "event", m.IfName, "op state not changed", m.operState)
-				continue
+			if linkUpdateEvent.Link.Attrs().OperState != m.operState {
+				m.handlePfEventForOpStateChange(linkUpdateEvent)
 			}
-			m.operState = newOpState
-			level.Info(m.Logger).Log("interface", "event", m.IfName, "op state changed", m.operState)
-			m.addrMutex.Lock()
-			if newOpState == netlink.OperDown {
-				for addr := range m.authenticatedAddrs {
-					err := m.deauthenticate(addr)
-					if err != nil {
-						level.Error(m.Logger).Log("interface", "addr", m.IfName, addr, "error deauthenticating addr", err)
-					}
-					m.deauthRequests[addr] = getCurrentTimestamp()
-				}
-				m.logEvent(kapi.EventTypeWarning, "interface is down")
-			} else if newOpState == netlink.OperUp {
-				for addr := range m.authenticatedAddrs {
-					delete(m.deauthRequests, addr)
-				}
-				m.logEvent(kapi.EventTypeNormal, "interface is up")
+			m.handlePfEventForVfVlanChange(linkUpdateEvent)
+		}
+	}
+}
+
+func (m *InterfaceMonitor) handlePfEventForVfVlanChange(linkUpdateEvent netlink.LinkUpdate) {
+	m.addrMutex.Lock()
+	defer m.addrMutex.Unlock()
+	pfLink, err := netlink.LinkByName(m.IfName)
+	if err != nil {
+		return
+	}
+	for _, vf := range pfLink.Attrs().Vfs {
+		if vfInfo, ok := m.PfInfo.VFs[vf.ID]; ok && vf.Vlan != trafficcontrol.ReservedVlan &&
```

Comment:
This kind of specialized knowledge feels like it belongs in sriov.go instead of here?